server/app.js

```diff
@@ -37,4 +39,176 @@ io.on('connection', (socket) => {
+    let profile = profiles.find(profile => profile.userID === msg.userID); // get their profile
+    let tableToJoin = tables.find(table => table.tableID === msg.tableID); // get the table they wish to join
+    if(canUserJoinTable(tableToJoin, profile)){// the user can join the table
+        let seatID = tableToJoin.addPlayerToTable(profile, socket.id); // the table will fill a seat based on the profile and send back the seat id
+        socket.emit('yourSeatID', seatID); // send him back his tableSeatID
+        io.emit('tableState', JSON.stringify(tableToJoin.getTableState())); // send to everyone the updated table state
+        if(tableToJoin.canAGameBegin() && !tableToJoin.tableActive){ // check if we can begin a game
+            tableToJoin.tableActive = true; // the table is now active
+            beginTheGame(io, tableToJoin);
+        }
+    }
+    else{
+        return; // the user can not join the table
+    }
+}
+
+
+
+
+
+
+
+
+
+// the functions for communication between the server and the poker.vue page
+function beginTheGame(io, table){ // function will begin the game with a time out of 5 seconds
+    setTimeout(function() { // in five seconds the game will begin
+        console.log('beginning the game');
+        table.beginTheGame(); // the game is set up
+        io.emit('tableState', JSON.stringify(table.getTableState())); // send to everyone the new table state
+        for(let i = 0; i < 6; i++){
+            let seat = table.tableSeats[i];
+            if(seat.socketID !== -1){
+                console.log('yes');
+                io.to(`${seat.socketID}`).emit('beginTheGame', JSON.stringify(seat.cards));
+            }
+        }
+        // now the player whos the current turn can send us a decision
+    }, 5000);
+}
+function turnDecision(io, socket, msg){ // each player when clicking the poker.vue buttons will trigger this function, it will assess whether or not the move was valid
+    if(tables[0].isItTheirTurn(msg.userID, msg.seatID, socket.id)){
+        // there is a limited number of actions a player can take based on their current state
+        // ask if they can make that move, if not then do nothing, later also disable buttons based on lack of options
+        let response = tables[0].playerAction(msg.action, msg.raiseToValue); // will assess the action and return a response
+        if(response){
+            io.emit('tableState', JSON.stringify(tables[0].getTableState()));
+        }
+        else{
+            console.log('bad move');
+        }
+        if(tables[0].showdown){// after each move we need to check if the table is ready for a showdown
+            console.log("show down begin");
+           beginShowingTheRemainingCommunityCards(io, tables[0]);
+        }
+    }
+}
+function beginShowingTheRemainingCommunityCards(io, table){ // the remaining community cards are revealed, one at a time (used if a premature show down is called)
+    let count = 5 - tables[0].communityCardsShown;
```

Comment:
Not sure how we will go about the magic numbers, but 5 is a magic number. 

To me, the context of the number is enough.